modules/twinkleblock.js

```diff
@@ -433,9 +433,37 @@ Twinkle.block.callback.change_action = function twinkleblockCallbackChangeAction
 			tags: true,
 			placeholder: 'Enter pages to block user from',
 			maximumSelectionLength: 10, // Software limitation [[phab:T202776]]
-			language: {
-				noResults: function() {
-					return 'No pages entered yet';
+			minimumInputLength: 1, // prevent ajax call when empty
+			createTag: function() { // prevent selection of arbitrary titles not returned by API
+				return null;
+			},
+			ajax: {
+				url: mw.util.wikiScript('api'),
+				dataType: 'json',
+				data: function(params) {
+					if (!params.term) {
+						return;
+					}
+					var title = new mw.Title(params.term);
```

Comment:
Per comment above, might be good to verify this is successful.

Reply:
Unlike above, this is the output returned by the API. Are there any legal titles that can't be parsed by `mw.Title` ? If so, I'd guess that needs to fixed MediaWiki-side. 

If there are such titles, I guess the fix would be return an empty object `{}` from the map (returning `null` or `undefined` results in errors).

Reply:
This comment was just in reference to https://github.com/azatoth/twinkle/pull/826#discussion_r371401798, although I thought I had a note about the one in `processResults` but 1. can't find it and 2. you dealt with it